x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/MountSearchableSnapshotRequest.java

```diff
@@ -0,0 +1,435 @@
+        }
+        if (mountedIndex == null) {
+            validationException = addValidationError("index name to mount is missing", validationException);
+        }
+        if (settings == null) {
+            validationException = addValidationError("settings are missing", validationException);
+        }
+        if (indexSettings == null) {
+            validationException = addValidationError("indexSettings are missing", validationException);
+        }
+        return validationException;
+    }
+
+    /**
+     * Sets the name of the snapshot.
+     *
+     * @param snapshot snapshot name
+     * @return this request
+     */
+    public MountSearchableSnapshotRequest snapshot(String snapshot) {
+        this.snapshot = snapshot;
+        return this;
+    }
+
+    /**
+     * Returns the name of the snapshot.
+     *
+     * @return snapshot name
+     */
+    public String snapshot() {
+        return this.snapshot;
+    }
+
+    /**
+     * Sets repository name
+     *
+     * @param repository repository name
+     * @return this request
+     */
+    public MountSearchableSnapshotRequest repository(String repository) {
+        this.repository = repository;
+        return this;
+    }
+
+    /**
+     * Returns repository name
+     *
+     * @return repository name
+     */
+    public String repository() {
+        return this.repository;
+    }
+
+    /**
+     * Sets the name of the index that should be restored from snapshot
+     */
+    public MountSearchableSnapshotRequest snapshotIndex(String indexName) {
+        this.snapshotIndex = indexName;
+        return this;
+    }
+
+    /**
+     * Returns index that should be restored from snapshot
+     */
+    public String snapshotIndex() {
+        return snapshotIndex;
+    }
+
+    /**
+     * Returns new index name after the snapshot has been mounted
+     */
+    public String mountedIndex() {
+        return mountedIndex;
+    }
+
+    /**
+     * If this parameter is set to true the operation will wait for completion of restore process before returning.
+     *
+     * @param waitForCompletion if true the operation will wait for completion
+     * @return this request
+     */
+    public MountSearchableSnapshotRequest waitForCompletion(boolean waitForCompletion) {
+        this.waitForCompletion = waitForCompletion;
+        return this;
+    }
+
+    /**
+     * Returns wait for completion setting
+     *
+     * @return true if the operation will wait for completion
+     */
+    public boolean waitForCompletion() {
+        return waitForCompletion;
+    }
+
+    /**
+     * Sets repository-specific restore settings.
+     * <p>
+     * See repository documentation for more information.
+     *
+     * @param settings repository-specific snapshot settings
+     * @return this request
+     */
+    public MountSearchableSnapshotRequest settings(Settings settings) {
```

Comment:
What are these exactly?

Reply:
I believe you can override repository-specific settings on a restore (things like speed), these are passed through to the settings that `RestoreSnapshotRequest` takes

Reply:
It looks like these settings are unused. I have removed them from the API here, and in the  future we can remove them from the usual restore process too.